compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -4931,7 +4931,15 @@ aloadHelper(TR::Node * node, TR::CodeGenerator * cg, TR::MemoryReference * tempM
             && constNode->isClassUnloadingConst())
       {
       uintptr_t value = constNode->getAddress();
-      TR::Instruction *unloadableConstInstr = generateRILInstruction(cg, TR::InstOpCode::LARL, node, tempReg, reinterpret_cast<void*>(value));
+      TR::Instruction *unloadableConstInstr = NULL;
+      if (cg->canUseRelativeLongInstructions(value))
+         {
+         unloadableConstInstr = generateRILInstruction(cg, TR::InstOpCode::LARL, node, tempReg, reinterpret_cast<void*>(value));
+         }
+      else
+         {
+         unloadableConstInstr = genLoadAddressConstantInSnippet(cg, node, value, tempReg, NULL, NULL, NULL, true);
+         }
```

Comment:
I think we should use `genLoadAddressConstant` here instead. Do you agree?

Reply:
If we use `genLoadAddressConstant`, then we will end up in this block in the routine:
```
if (node->isClassUnloadingConst())
      {
      uintptr_t value = node->getAddress();
      TR::Instruction *unloadableConstInstr = generateRILInstruction(cg, TR::InstOpCode::LARL, node, targetRegister, reinterpret_cast<void*>(value));
      TR_OpaqueClassBlock* unloadableClass = NULL;
      if (node->isMethodPointerConstant())
         {
         comp->getStaticMethodPICSites()->push_front(unloadableConstInstr);
         }
      else
         {
         comp->getStaticPICSites()->push_front(unloadableConstInstr);
         }
      return unloadableConstInstr;
      }
```

So this wouldn't sold the problem. We would have to implement the same type of solution inside here.

Reply:
`genLoadAddressConstant` looks like it's targeted towards `aconst` nodes, while `aloadHelper` here is for `aload` nodes. I think we should leave the checks inside these methods as is, and can possibly refactor the code inside the blocks into a common routine. I'll look into this.

Reply:
Yeah you're right. Upon closer inspection it looks like we can't handle what we want here by calling `genLoadAddressConstant`. I'm ok with what you have then.

Reply:
I think the code in `genLoadAddressConstant ` might be incorrect for remote compilations. We'll need to add a similar `canUseRelativeLongInstructions` check in there as well to make sure we don't generate PC-relative instructions.

Reply:
Added the check here: e855952